### PR TITLE
codemirror blob view: First pass on porting blob tests

### PR DESCRIPTION
**Pull request by fkling**
This commit ports the non-skipped tests from `blob-viewer.test.ts` to
test them against the CodeMirror implementation.
**Note**: Most of it is just copied. I only had to make some minor
adjustments to make it work with CodeMirror's DOM structure.
~I also copied the `skip`ed tests with the intention to reactivate them.~

~The most significant change is probably how "selections" work. Instead
of simply clicking on a line (which did not work) I extended the
existing editor API to allow us to specify the position of the text
cursor.~
~That only works because currently the view is configured to show/have a
cursor. If we are not keeping this functionality we have to find another
approach.~

I dropped two of the skipped test from `blob-viewer.test.ts`:
- One didn't contain any actual content
- The other one relates to the reference panel and I feel like that one
  should probably live elsewhere.

The skipped test for hovers didn't contain any logic, I added some.

I added a couple of tests to test line selection via lines and lines numbers,
shift clicking and dragging.

**Note**: This test cannot be run in debug mode without further changes
because it relies overwriting the JavaScript context object which doesn't
work in debug mode atm.

## Test plan

Run test locally:

```
INTEGRATION_TESTS=1 sg start web-standalone
```

and

```
yarn test-integration:debug client/web/src/integration/codemirror-blob-viewer.test.ts
```      

CI passes for new test.

## App preview:

- [Web](https://sg-web-fkling-blob-tests.onrender.com)
- [Storybook](https://5f0f381c0e50750022dc6bf7-smhzxcopbb.chromatic.com)

Check out the [client app preview documentation](https://docs.sourcegraph.com/dev/how-to/client_pr_previews) to learn more.

### CODE REVIEW

File: client/web/src/integration/codemirror-blob-viewer.test.ts [hunk 1]
```diff
@@ -0,0 +1,982 @@
...
+
+            // Count child nodes of existing portals
+            assert.strictEqual(
+                await driver.page.evaluate(
+                    (selector: string) => document.querySelector(selector)?.childElementCount,
+                    lineDecorationAt(2)
+                ),
+                1,
+                'Expected line 2 to have 1 decoration'
+            )
```
Comment (fkling) [inline client/web/src/integration/codemirror-blob-viewer.test.ts:518]: This part contains lines that I actually changed.

File: client/web/src/integration/codemirror-blob-viewer.test.ts [hunk 2]
```diff
@@ -0,0 +1,982 @@
...
+                'Expected line 1 to not have a decoration attachment on test.ts (second visit)'
+            )
+            assert(
+                await driver.page.$(lineDecorationAt(2)),
+                'Expected line 2 to have a decoration attachment on test.ts (second visit)'
+            )
+            assert(
+                !(await driver.page.$(lineDecorationAt(3))),
+                'Expected line 3 to not have a decoration attachment on test.ts (second visit)'
+            )
```
Comment (fkling) [inline client/web/src/integration/codemirror-blob-viewer.test.ts:752]: As does this one.

**Review by umpox — APPROVED** [review]
> I also copied the skiped tests with the intention to reactivate them.

Would be a big win as these tests also contained `percySnapshot`s, and I think we have very little visual regression coverage on the blob pages :)

**Review by umpox — APPROVED** [review]
Nice!

**Comment by fkling** [comment]
Accidentally added a couple of wrong files (which are now removed), sorry for the ping @limitedmage

**Pull request merged**